packages/angular/ssr/src/manifest.ts

```diff
@@ -10,7 +10,20 @@ import type { SerializableRouteTreeNode } from './routes/route-tree';
 import { AngularBootstrap } from './utils/ng';
 
 /**
- * Represents of a server asset stored in the manifest.
+ * Utility type that replaces all properties of type `Record<K, V>` with `ReadonlyMap<K, V>`.
+ */
+type TransformRecordWithReadonlyMap<T> = {
+  [K in keyof T]: T[K] extends (...args: unknown[]) => unknown // Check if it's a function
+    ? T[K] // Leave function properties untouched
+    : T[K] extends Readonly<Record<string, unknown>> // Check if it's a Readonly Record
+      ? T[K] extends Readonly<Record<infer RK, infer RV>>
+        ? ReadonlyMap<RK, RV> // Convert Readonly Record to ReadonlyMap
+        : never
+      : T[K]; // Leave other properties unchanged
+};
```

Comment:
```suggestion
type TransformRecordWithReadonlyMap<T> = {
  [K in keyof T]: T[K] extends (...args: unknown[]) => unknown // Check if it's a function
    ? T[K] // Leave function properties untouched
    : T[K] extends Readonly<Record<infer RK, infer RV>>
        ? ReadonlyMap<RK, RV> // Convert Readonly Record to ReadonlyMap
        : T[K]; // Leave other properties unchanged
};
```

Suggestion: Might be moot if we drop the `Map`, but can we get rid of the double `extends Readonly<...>` check? I think we can `infer` at the same time?

Reply:
I tried this, but it does not work as expected